core/tauri/src/api/file/extract.rs

```diff
@@ -119,9 +119,31 @@ impl<'a> Extract<'a> {
         let mut archive = zip::ZipArchive::new(source)?;
         for i in 0..archive.len() {
           let mut file = archive.by_index(i)?;
-          let path = into_dir.join(file.name());
-          let mut output = fs::File::create(path)?;
-          io::copy(&mut file, &mut output)?;
+          // Decode the file name from raw bytes instead of using file.name() directly.
+          // file.name() uses String::from_utf8_lossy() which may return messy characters
+          // such as: τê▒Σ║ñµÿô.app/, that does not work as expected.
+          // Here we require the file name must be a valid UTF-8.
+          let file_name = String::from_utf8(file.name_raw().to_vec())?;
+          let out_path = into_dir.join(&file_name);
+          if file_name.ends_with('/') || file_name.ends_with('\\') {
+            fs::create_dir_all(&out_path)?;
+          } else {
+            if let Some(out_path_parent) = out_path.parent() {
+              if !out_path_parent.exists() {
```

Comment:
No need to check if parent exists or not, just use `fs::create_dir_all()` directly, it will handle it.

Reply:
It's necessary.  Because some zip archives may not contain directory entries.

Reply:
that's not what I meant. 
```rs
// at this point we are dealing with a file
if let Some(out_path_parent) = out_path.parent() {
	if !out_path_parent.exists() {
	}
}
```

Reply:
that's not what I meant. 
```rs
// at this point we are dealing with a file
if let Some(out_path_parent) = out_path.parent() {
	// at this point the out path has a parent directory that needs to be created first
	// here you call `fs::create_dir_all()` only if this parent doesn't exits
	// what I am saying is you don't need to check if it exists or not because `fs::create_dir_all()` will handle that. 
	if !out_path_parent.exists() {
		fs::create_dir_all(&out_path_parent)?;
	}
}
```
you can just do this 
```rs
if let Some(out_path_parent) = out_path.parent() {
	fs::create_dir_all(&out_path_parent)?;
}
```

Reply:
Done.